server/chainquery/queries/claimQueries.js

```diff
@@ -0,0 +1,184 @@
+    shortId = longId.substring(0, shortIdLength);
+    possibleMatches = possibleMatches.filter(element => {
+      return (element.claim_id && (element.claim_id.substring(0, shortIdLength) === shortId));
+    });
+  }
+  return shortId;
+};
+
+const isLongClaimId = (claimId) => {
+  return (claimId && (claimId.length === 40));
+}
+
+const isShortClaimId = (claimId) => {
+  return (claimId && (claimId.length < 40));
+}
+
+export default (db, table) => ({
+
+  getShortClaimIdFromLongClaimId: async (claimId, claimName) => {
+    logger.debug(`claim.getShortClaimIdFromLongClaimId for ${claimName}#${claimId}`);
+    return await table.findAll({
+      where: { name: claimName },
+      order: [['height', 'ASC']],
+    }).then(result => {
+      if(result.length === 0) {
+        throw new Error('No claim(s) found with that claim name');
+      }
+
+      return returnShortId(result, claimId);
+    });
+  },
+
+  getAllChannelClaims: async (channelClaimId) => {
+    logger.debug(`claim.getAllChannelClaims for ${channelClaimId}`);
+    return await table.findAll({
+      where: { publisher_id: channelClaimId },
+      order: [['height', 'DESC']],
+      raw  : true,  // returns an array of only data, not an array of instances
+    })
+    .then(channelClaimsArray => {
+      if(channelClaimsArray.length === 0) {
+        return null;
+      }
+
+      return channelClaimsArray;
+    })
+  },
+
+  getClaimIdByLongChannelId: async (channelClaimId, claimName) => {
+    logger.debug(`finding claim id for claim ${claimName} from channel ${channelClaimId}`);
+    return await table.findAll({
+      where: { name: claimName, publisher_id: channelClaimId },
+      order: [['id', 'ASC']],
+    })
+    .then(result => {
+      switch (result.length) {
+        case 0:
+          return null;
+        case 1:
+          return result[0].claim_id;
+        default:
+          // Does this actually happen??? (from converted code)
+          logger.warn(`${result.length} records found for "${claimName}" in channel "${channelClaimId}"`);
+          return result[0].claim_id;
+      }
+    });
+  },
+
+  validateLongClaimId: async (name, claimId) => {
+    return await table.findOne({
+      where: {
+        name,
+        claim_id: claimId,
+      },
+    }).then(result => {
+      if (!result) {
+        return false;
+      }
+      return claimId;
+    });
+  },
+
+  getLongClaimIdFromShortClaimId: async (name, shortId) => {
+    return await table.findAll({
+      where: {
+        name,
+        claim_id: {
+          [sequelize.Op.like]: `${shortId}%`,
+        }},
+      order: [['height', 'ASC']],
+    })
+    .then(result => {
+      if(result.length === 0) {
+        return null;
+      }
+
+      return result[0].claim_id;
+    });
+  },
+
+  getTopFreeClaimIdByClaimName: async (name) => {
+    return await table.findAll({
+      // TODO: Limit 1
+      where: { name },
+      order: [['effective_amount', 'DESC'], ['height', 'ASC']],
```

Comment:
So not sure we should use this. The `bid_state` = `Controlling` is better. This information comes directly from the blockchain.